Bug - Can't make a backup - FOREGROUND_SERVICE_DATA_SYNC

When I try to create a backup, the app creates a 0B file, then crashes. I don't have syncing enabled.

I'm on a Pixel 7 running Graphene OS. I couldn't make it work by disabling any security feature, turning on compatibility mode or by clearing the cache.

Quillpad v1.4.14 / 20

Crash report

```
type: crash  
osVersion: google/panther/panther:14/AP1A.240305.019.A1/2024032100:user/release-keys  
package: io.github.quillpad:20  
process: io.github.quillpad  
processUptime: 67865 + 222 ms  
installer: dev.imranr.obtainium  
  
java.lang.SecurityException: Starting FGS with type dataSync callerApp=ProcessRecord{ea90083 23540:io.github.quillpad/u0a271} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_DATA_SYNC]   
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3191)  
	at android.os.Parcel.createException(Parcel.java:3175)  
	at android.os.Parcel.readException(Parcel.java:3151)  
	at android.os.Parcel.readException(Parcel.java:3093)  
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6876)  
	at android.app.Service.startForeground(Service.java:777)  
	at xa.d.s(SourceFile:245)  
	at l8.a.k(Unknown Source:8)  
	at b9.n0.run(Unknown Source:100)  
	at g9.i.run(Unknown Source:12)  
	at i9.j.run(Unknown Source:2)  
	at i9.a$a.run(SourceFile:82)  
	Suppressed: b9.l0: [e2{Cancelling}@f70e056, Dispatchers.IO]  
Caused by: android.os.RemoteException: Remote stack trace:  
	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2685)  
	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2375)  
	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1704)  
	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13768)  
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3438)  
```

Logs

```
type: logcat
osVersion: google/panther/panther:14/AP1A.240305.019.A1/2024032100:user/release-keys
packageName: io.github.quillpad:20
buffers: main,system,crash,events
level: verbose


1711395539.107 31709 31709 D CompatibilityChangeReporter: Compat change id reported: 280005585; UID 10271; state: DISABLED
--------- switch to events
1711395539.298 31709 31709 I wm_on_paused_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performPause,time=1ms]
--------- switch to main
1711395539.316 31709 31709 D VRI[MainActivity]: visibilityChanged oldVisibility=true newVisibility=false
1711395539.325 31709 31709 D VRI[MainActivity]: Not drawing due to not visible
--------- switch to events
1711395539.329 31709 31709 I wm_on_stop_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=STOP_ACTIVITY_ITEM,time=1ms]
1711395725.192 31709 31709 I wm_on_restart_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performRestart,time=0ms]
1711395725.224 31709 31709 I wm_on_start_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=handleStartActivity,time=31ms]
1711395725.229 31709 31709 I wm_on_resume_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=RESUME_ACTIVITY,time=0ms]
1711395725.229 31709 31709 I wm_on_top_resumed_gained_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=topWhenResuming]
--------- switch to main
1711395725.346 31709 31709 D VRI[MainActivity]: Start draw after previous draw not visible
1711395729.874 31709 31709 D CompatibilityChangeReporter: Compat change id reported: 171228096; UID 10271; state: ENABLED
--------- switch to events
1711395731.605 31709 31709 I wm_on_top_resumed_lost_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=topStateChangedWhenResumed]
1711395731.605 31709 31709 I wm_on_paused_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performPause,time=1ms]
--------- switch to main
1711395732.256 31709 31709 D VRI[MainActivity]: visibilityChanged oldVisibility=true newVisibility=false
1711395732.270 31709 31739 D HWUI    : endAllActiveAnimators on 0xb400bf980eee3a00 (RippleDrawable) with handle 0xb400be6e7b02b000
1711395732.277 31709 31709 D VRI[MainActivity]: Not drawing due to not visible
--------- switch to events
1711395732.280 31709 31709 I wm_on_stop_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=STOP_ACTIVITY_ITEM,time=1ms]
--------- switch to main
1711395733.474 31709 31709 D CompatibilityChangeReporter: Compat change id reported: 78294732; UID 10271; state: ENABLED
--------- switch to events
1711395733.475 31709 31709 I wm_on_restart_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performRestart,time=0ms]
1711395733.480 31709 31709 I wm_on_start_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=handleStartActivity,time=5ms]
1711395733.485 31709 31709 I wm_on_activity_result_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=ACTIVITY_RESULT]
1711395733.485 31709 31709 I wm_on_resume_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=RESUME_ACTIVITY,time=0ms]
1711395733.486 31709 31709 I wm_on_top_resumed_gained_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=topWhenResuming]
--------- switch to main
1711395733.501 31709 31709 D VRI[MainActivity]: Start draw after previous draw not visible
--------- switch to crash
1711395733.570 31709 31771 E AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-8
1711395733.570 31709 31771 E AndroidRuntime: Process: io.github.quillpad, PID: 31709
1711395733.570 31709 31771 E AndroidRuntime: java.lang.SecurityException: Starting FGS with type dataSync callerApp=ProcessRecord{b4d310c 31709:io.github.quillpad/u0a271} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_DATA_SYNC]
1711395733.570 31709 31771 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3191)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3175)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3151)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3093)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6876)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:777)
1711395733.570 31709 31771 E AndroidRuntime: 	at xa.d.s(SourceFile:245)
1711395733.570 31709 31771 E AndroidRuntime: 	at l8.a.k(Unknown Source:8)
1711395733.570 31709 31771 E AndroidRuntime: 	at b9.n0.run(Unknown Source:100)
1711395733.570 31709 31771 E AndroidRuntime: 	at g9.i.run(Unknown Source:12)
1711395733.570 31709 31771 E AndroidRuntime: 	at i9.j.run(Unknown Source:2)
1711395733.570 31709 31771 E AndroidRuntime: 	at i9.a$a.run(SourceFile:82)
1711395733.570 31709 31771 E AndroidRuntime: 	Suppressed: b9.l0: [e2{Cancelling}@9123789, Dispatchers.IO]
1711395733.570 31709 31771 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
1711395733.570 31709 31771 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2685)
1711395733.570 31709 31771 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2375)
1711395733.570 31709 31771 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1704)
1711395733.570 31709 31771 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13768)
1711395733.570 31709 31771 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3438)
1711395733.570 31709 31771 E AndroidRuntime:
--------- switch to events
1711395733.577 31709 31709 I wm_on_top_resumed_lost_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=topStateChangedWhenResumed]
1711395733.579 31709 31709 I wm_on_paused_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performPause,time=0ms]
--------- switch to main
1711395734.193 31709 31709 D VRI[MainActivity]: visibilityChanged oldVisibility=true newVisibility=false
1711395734.209 31709 31709 D VRI[MainActivity]: Not drawing due to not visible
--------- switch to events
1711395734.214 31709 31709 I wm_on_stop_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=LIFECYCLER_STOP_ACTIVITY,time=0ms]
1711395734.218 31709 31709 I wm_on_destroy_called: [Token=255600734,Component Name=org.qosp.notes.ui.MainActivity,Reason=performDestroy,time=3ms]
--------- switch to main
1711395734.218 31709 31709 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda11@221aec4
```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -85,4 +85,11 @@
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
   <uses-permission android:name="android.permission.SCHEDULE_EXACT_ALARM" />
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC"
+                   android:minSdkVersion="34"/>
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MICROPHONE"
+                   android:minSdkVersion="34"/>
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK"
+                   android:minSdkVersion="34"/>
+
 </manifest>

